telegram-bot-api/StatsJson.h

```diff
@@ -0,0 +1,285 @@
+//
+// Copyright Luckydonald (tdlight-telegram-bot-api+code@luckydonald.de) 2020
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+#pragma once
+
+#include "telegram-bot-api/Query.h"
+
+#include "td/db/TQueue.h"
+
+#include "td/net/HttpOutboundConnection.h"
+#include "td/net/HttpQuery.h"
+#include "td/net/SslStream.h"
+
+#include "td/actor/actor.h"
+#include "td/actor/PromiseFuture.h"
+
+#include "td/utils/common.h"
+#include "td/utils/Container.h"
+#include "td/utils/FloodControlFast.h"
+#include "td/utils/HttpUrl.h"
+#include "td/utils/JsonBuilder.h"
+#include "td/utils/List.h"
+#include "td/utils/port/IPAddress.h"
+#include "td/utils/port/SocketFd.h"
+#include "td/utils/Slice.h"
+#include "td/utils/Status.h"
+#include "td/utils/VectorQueue.h"
+#include "td/utils/utf8.h"
+
+#include <atomic>
+#include <functional>
+#include <memory>
+#include <set>
+#include <tuple>
+#include <unordered_map>
+
+namespace telegram_bot_api {
+
+class JsonStatsSize : public td::Jsonable {
+ public:
+  explicit JsonStatsSize(td::uint64 size) : size_(size) {
+  }
+  void store(td::JsonValueScope *scope) const {
+    auto object = scope->enter_object();
+    object("bytes", td::JsonLong(size_));
+
+    // Now basically td::format::as_size(...), but without need for an StringBuilder.
+    struct NamedValue {
+      const char *name;
+      td::uint64 value;
+    };
+
+    static constexpr NamedValue sizes[] = {{"B", 1}, {"KB", 1 << 10}, {"MB", 1 << 20}, {"GB", 1 << 30}};
+    static constexpr size_t sizes_n = sizeof(sizes) / sizeof(NamedValue);
+
+    size_t i = 0;
+    while (i + 1 < sizes_n && size_ > 10 * sizes[i + 1].value) {
+      i++;
+    }
+    object("human_readable", td::to_string(size_ / sizes[i].value) + sizes[i].name);
+  }
+
+ private:
+  const td::uint64 size_;
+};
+
+class JsonStatsMem : public td::Jsonable {
+ public:
+  explicit JsonStatsMem(const td::MemStat mem_stat) : mem_stat_(mem_stat) {
+  }
+  void store(td::JsonValueScope *scope) const {
+    auto object = scope->enter_object();
+    object("resident_size", JsonStatsSize(mem_stat_.resident_size_));
+    object("resident_size_peak", JsonStatsSize(mem_stat_.resident_size_peak_));
+    object("virtual_size", JsonStatsSize(mem_stat_.virtual_size_));
+    object("virtual_size_peak", JsonStatsSize(mem_stat_.virtual_size_peak_));
+  }
+
+ private:
+  const td::MemStat mem_stat_;
+};
+
+
+class JsonStatsCpuItem : public td::Jsonable {
```

Comment:
i can't find any usage of this class, remove if unused

Reply:
Ah yes. Good catch. Was replaced by `JsonStatsCpu` later.